src/objectExplorer/objectExplorerService.ts

```diff
@@ -160,12 +174,14 @@ export class ObjectExplorerService {
             nodePath: node.nodePath
         };
         this._expandParamsToPromiseMap.set(expandParams, promise);
+        this._expandParamsToTreeNodeInfoMap.set(expandParams, node);
```

Comment:
Instead of 2 separate maps I have a slight preference to just a single map where the value contains the promise and node. Just less to keep track of when deleting/updating the map.